pydantic/_internal/_validators.py

```diff
@@ -57,23 +57,19 @@ def sequence_validator(
     Validator for `Sequence` types, isinstance(v, Sequence) has already been called.
     """
     value_type = type(__input_value)
+
+    # We don't accept any plain string as a sequence
+    # Relevant issue: https://github.com/pydantic/pydantic/issues/5595
+    if issubclass(value_type, (str, bytes)):
+        raise PydanticCustomError(
+            'sequence_str_str_type', f'`{value_type.__name__}` instances are not allowed as a Sequence value'
```

Comment:
```suggestion
            'sequence_str_type', f"'{type}' instances are not allowed as a Sequence value', {'type': value_type.__name__}
```
Or your could give a different error for `str` and `bytes` which would avoid a confusing error if you pass a subclass of str or bytes here.

The point is that there should be a one-to-one mapping between `error_type` and `message_template` to allow translations of error messages, hence we use context for variables, not dynamic message templates.

Reply:
also ussing single quotes for variables matches the style in other error messages.

Reply:
Regarding the error type name, I'd rather keep `sequence_str_str_type` meaning it's `str_type` error for `sequence_str` type. So, there could be `sequence_int_str_type` as well for example.